sessionctx/stmtctx/stmtctx.go

```diff
@@ -393,6 +409,7 @@ func (sc *StatementContext) ResetForRetry() {
 	sc.mu.Unlock()
 	sc.TableIDs = sc.TableIDs[:0]
 	sc.IndexIDs = sc.IndexIDs[:0]
+	sc.ReSetNowTs()
```

Comment:
I think we should not reset when retrying. Transaction retry is an automatic implicit behavior. It is possible that current_timestamp has been returned to the client. If it is reset here, the user may be bothered. For example,

```sql
begin;
insert xxx current_timestamp;
select now(); -- The user might think this value is bigger than the previous insert.
commit; -- It is better do not reset current here, if the retry is needed.
```

Reply:
PTAL @lysu